util/progress/progressui/colors.go

```diff
@@ -0,0 +1,142 @@
+package progressui
+
+import (
+	"encoding/csv"
+	"strconv"
+	"strings"
+
+	"github.com/morikuni/aec"
+	"github.com/sirupsen/logrus"
+)
+
+var termColorMap = map[string]aec.ANSI{
+	"default": aec.DefaultF,
+
+	"black":   aec.BlackF,
+	"blue":    aec.BlueF,
+	"cyan":    aec.CyanF,
+	"green":   aec.GreenF,
+	"magenta": aec.MagentaF,
+	"orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 140, 0)),
+	"red":     aec.RedF,
+	"white":   aec.WhiteF,
+	"yellow":  aec.YellowF,
+
+	"light-black":   aec.LightBlackF,
+	"light-blue":    aec.LightBlueF,
+	"light-cyan":    aec.LightCyanF,
+	"light-green":   aec.LightGreenF,
+	"light-magenta": aec.LightMagentaF,
+	"light-orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 165, 0)),
+	"light-red":     aec.LightRedF,
+	"light-white":   aec.LightWhiteF,
+	"light-yellow":  aec.LightYellowF,
+}
+
+func setUserDefinedTermColors(colorsEnv string) {
+	fields := readBuildkitColorsEnv(colorsEnv)
+	if fields == nil {
+		return
+	}
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		if len(parts) != 2 {
+			logrus.Warnf("Could not parse BUILDKIT_COLORS component: %s", strings.Join(parts, "="))
+			continue
+		}
+		if strings.Contains(parts[1], "=") {
```

Comment:
Not quite sure what case this catches? I think this would get caught anyways a bit further down, since it isn't a valid rgb or per-defined color.

Reply:
This catches "warning=yellow=red" or something else. You are right that it will get caught later, but this error message points more directly at the problem since it will print the whole field and say that it is not parseable. I'd prefer to leave this, instead of falling down to the unknown color error message.